scriptmodules/admin/setup.sh

```diff
@@ -175,7 +175,9 @@ function updatescript_setup()
     fi
     popd
     rp_callModule runcommand install
-    printMsgs "dialog" "Fetched the latest version of the RetroPie Setup script. You need to restart the script."
+    printMsgs "dialog" "Fetched the latest version of the RetroPie Setup script. Reloading the updated script now."
+    "$scriptdir/retropie_setup.sh"
+    exit 0 
 }
```

Comment:
This will launch the script again, whilst the old script is still loaded (the original script wont exit until the new one exits) - it would probably be better to use "exec" or something to avoid that

http://wiki.bash-hackers.org/commands/builtin/exec